### ISSUE TITLE
Why 0xE0FF, 0xEFFD, 0xF000 in Latin name list?

### ISSUE THREAD

**Issue by MalcolmWooden** [googlefonts/tools#45]
What is the significance of these three PUA characters in the basic Latin name file? (https://github.com/googlefonts/tools/blob/master/encodings/latin_unique-glyphs.nam) Should these really be included with new fonts?

**Comment by madig** [comment]
And if so, what should they contain, what spacing/advance width shoul they have?

**Comment by EbenSorkin** [comment]
I agree. Unicode E0FF, EFFD and F000 should probably be deleted.

**Comment by EbenSorkin** [comment]
BTW - These are in the Google Core list.

**Comment by davelab6** [comment]
I forget exactly, but one is there for 'foundry glyphs', akin to the Apple logo glyph (`0xF8FF` https://superuser.com/questions/1205451/how-can-i-display-the-uf8ff-apple-logo-character-on-windows) and the other 2 were for hinting glyphs that are found in the Deja Vu fonts (which GF doesn't host) and Ubuntu (which GF does). 

Indeed, https://github.com/matplotlib/matplotlib/blob/master/tools/subset.py#L202-L204 is a very old py script that has comments about this stuff:

    latin += [0xe0ff] # PUA: Font logo
    latin += [0xeffd] # PUA: Font version number
    latin += [0xf000] # PUA: font ppem size indicator: run `ftview -f 1255 10 Ubuntu-Regular.ttf` to see it in action!

I made a test page and indeed the actual `latin` subset served by GF no longer includes these characters, so they should be removed from the "GF 2016" glyph sets and the `latin.nam` file:

<img width="1012" alt="screen shot 2018-05-18 at 13 17 32" src="https://user-images.githubusercontent.com/261579/40255990-ddc3c6f6-5a9d-11e8-8a0c-e4c17e8b9ade.png">

However, when using the `&text=` API feature, you can request these PUA chars, and see the funky hinting code and logo in action :)

<img width="1036" alt="screen shot 2018-05-18 at 13 18 55" src="https://user-images.githubusercontent.com/261579/40256060-19b1066a-5a9e-11e8-80fd-26a8854210da.png">

Demo page link: https://jsbin.com/guzevoq/edit?html,output

**Comment by madig** [comment]
Why not remove them? They aren't really core, more like fringe special cases.

**Comment by davelab6** [comment]
@RosaWagner is leading an effort to define a new "GF Core" set for designers.

**Comment by RosaWagner** [comment]
We need to define what is `Core`. On one hand it could be the basic characters that you can type easily with a keyboard (with crtl/cmd/alt/shift + key) from countries using latin script — meaning covering `MacRoman` and `Windows1252` glyphsets. In that case, we should keep `fraction sign` and ditch `division slash`.

But the `fraction sign` is mostly used through the fraction feature which substitute `slash` by `fraction sign`. If the Latin Core set is not containing any `dnom` or `numr` figures to make that feature works, then I don't see a big reason of keeping it. 

Now, this is a list of characters which are missing comparing to MacRoman and Windows1252, ie these characters are also accessible through a keyboards from various countries:

```
- Lslash
- Scaron
- Ydieresis
- Zcaron
- lslash
- scaron
- zcaron
- fi
- fl
- Delta
- Omega
- pi
- florin
- trademark
- dagger
- daggerdbl
- notequal
- greaterequal
- lessequal
- approxequal
- infinity
- integral
- product
- summation
- radical
- micro
- partialdiff
- perthousand
- lozenge
- dotaccent
- hungarumlaut
- caron
- breve
- ogonek
```

I understand why they are not included in GF Latin Core. A lot of glyphs in there are kind of legacy unused stuff. Also we don't want to overload the Latin Core set because (on the other hand) it is actually the set supposed to be used by fonts which don't have Latin as a main/default script.

Although, these signs allow basic mathematics, and should be easy to add to the font once you already designed the symbols from the actual Core set — and they are not specific to latin script.

```
- notequal
- greaterequal
- lessequal
- approxequal
- perthousand
```

Finally, GF Latin Core is misleading designers by making them do the legacy accents and not the comb (zero-width) accents. Since we ask designers to have minimal consistency between the accented glyphs and the mark_to_mark feature to allow automatic composition of accented letters, it seems important to recommend these glyphs as well:

```
- dieresiscomb
- gravecomb
- acutecomb
- circumflexcomb
- ringcomb
- tildecomb
- macroncomb
- cedillacomb

```

**Comment by EbenSorkin** [comment]
Having followed along with this now I do think these all seem like sound
recommendations.

-e.

**Comment by vv-monsalve** [comment]
> it could be the basic characters that you can type easily with a keyboard (with crtl/cmd/alt/shift + key) from countries using latin script — meaning covering MacRoman and Windows1252 glyphsets.

Agree, although `MacRoman` actually includes fractions such as ½,  ¼, ¾, and the Core encoding is requiring one, two and three superior. So I think it is a good reason to keep the `fraction` but perhaps to also require at least one, two, three and four `dnom` and `numrs`

> Latin Core set because (on the other hand) it is actually the set supposed to be used by fonts which don't have Latin as a main/default script

```
Google Fonts expects that fonts in its collection support at least the minimal
set of characters defined in the GF-latin-core glyph-set.
```
As per the check is still the minimum set for many Latin fonts (especially the not commissioned ones). So ensuring the covering of minimum Latin standards including a couple more glyphs could be good

```
- Lslash
- Scaron
- Ydieresis
- Zcaron
- lslash
- scaron
- zcaron
- fi
- fl
- notequal
- greaterequal
- lessequal
- approxequal
- micro (already included in Core)
- dotaccent + comb
- hungarumlaut + comb
- caron + comb
- breve + comb
- ogonek + comb
```

Keeping the legacy marks + the comb ones would be also good for backwards compatibility.

**Comment by khaledhosny** [comment]
If `Core` set must be included in every font regardless of its intended script/language, then the set must be conservative and have a clear rational for every symbol made required, not just because it was in some legacy encoding.

From my point of view, this list should include only non-control characters from Unicode Basic Latin (AKA ASCII) block and nothing more. 

Fonts that are mainly Latin (or Latin is one of their main scripts) can have a more extended coverage.

**Comment by twardoch** [comment]
> basic characters that you can type easily with a keyboard (with crtl/cmd/alt/shift + key) from countries using latin script 

1. Which operating system does this refer to? 
2. Which software keyboard layout does this refer to? 
3. Which countries does this refer to?

**Comment by twardoch** [comment]
> If `Core` set must be included in every font regardless of its intended script/language, then the set must be conservative and have a clear rational for every symbol made required, not just because it was in some legacy encoding.

Yep. Unicode was widely introduced with Windows XP in 2001 = before quite a few people who use computers were born. 

> From my point of view, this list should include only non-control characters from Unicode Basic Latin (AKA ASCII) block and nothing more.

Yep.

> Fonts that are mainly Latin (or Latin is one of their main scripts) can have a more extended coverage.

Yep. Why include glyphs for a 300-thousand Icelandic language but not for Polish which has over 100x more users? Because some engineers (who are long-retired now) decided that in the 1980s when the Soviet Union still existed and character sets had to be limited to 8-bit? 

I think this is an odd path to go.

**Comment by twardoch** [comment]
I live in a European country that uses the Latin script and I can easily type `ą` with Alt+a on both macOS and Windows, using the »Polish Pro[grammars]« keyboard layout. Is `ą` in the Core set? If not — why?

**Comment by twardoch** [comment]
https://github.com/googlefonts/gftools/blob/main/Lib/gftools/encodings/GF%20Glyph%20Sets/GF-latin-core_unique-glyphs.nam includes characters like thorn that must be drawn separately, but does not include various precomposed characters that can be added automatically, but does includes some. Maybe I’m so old that I don't remember the times when this may have been a rational choice (= before Unicode and OpenType). But in 2016? In 2021? 😀

**Comment by twardoch** [comment]
Khaled is right that ASCII still occupies a special place. Not because the English language but because the ASCII characters are also widely used in various code-like functions. Most programming languages use ASCII. Most usernames and passwords are ASCII. E-mail addresses are mostly ASCII. So ASCII is the today’s equivalent to digits. ASCII is used by users of other scripts. So it does make sense to make them core. Even font menu names are usually ASCII and it does make sense to include characters in fonts that some apps will use to render the font's name. Most apps will render the font name in a system font if the font doesn't include the necessary characters, but that may be confusing. 

But the »Latin-1« bits of Windows 1252, MacRoman, ISO 8859-1 are just very arbitrary selections of characters made to fit a technical limitation of 256, which no longer applies anywhere, certainly not in environments that would use new-made fonts. 

Petrifying the memory of 44 years of communist regime in some European countries is not a good idea, I think, especially since nearly that much time passed since these countries stopped being communist 😀

The U.S. English keyboard layout is just one among many, and basing selections on what can be accessed on that layout in one particular operating system is not a good approach either, I think. 

Why a »Latin« font would be required to include letters for Portuguese or Icelandic but not Polish or Latvian, when these letters can be made equally easy — that’s something many users these days no longer understand. 

So — I’d be for a »core« set that's ASCII, for technical reasons. Anything above should be based on contemporary linguistic and audience-based criteria, not ancient history.

**Comment by twardoch** [comment]
Perhaps we should start by something like https://en.wikipedia.org/wiki/List_of_countries_by_number_of_Internet_users — then map these numbers to predominant scripts and languages in these countries, list characters needed by these countries, and use a simple "penalty" score for whether a glyph for that language can be made as a composite or needs to be drawn/adjusted. Then we could make a cut-off at some point, but not for “core”, I think (which, as Khaled said, just be just ASCII), but for the other thing.

**Comment by twardoch** [comment]
For a more mixed scoring, we may use the number of internet users divided by 1000 plus the GDP in USD millions divided by 10 from https://en.wikipedia.org/wiki/List_of_countries_by_GDP_(nominal) — that would use 50% of the population and 50% of the economic strength, in comparable units. No matter what mixed scoring we apply to identify the “importance” of the markets, they’ll never be objective, but they can guide decision-making better than what one or two Apple/Microsoft engineers 40 years ago decided :)

**Comment by RosaWagner** [comment]
Thanks you all for your contribution. 

We all agree that Mac Roman and Windows 1252 is not to strictly follow because old and not adapted to current usage. We can proceed.

I understand 2 things:
1. Fonts which don’t have Latin has main script should be covering ascii. Good idea.
2. core set for Latin fonts as main script is still to define then. This means, what to remove, and what to add to the existing Latin Core glyph set (I noted what has been proposed so far)

Nota bene: For this to move effectively forward, we need a **concrete list of glyphs with a synthetic and short comment to explain the reason.**

**Comment by davelab6** [comment]
Over the last few years this has come up in the context of defining a concept glyph set for initial milestones in the commissioned projects that I had more of an art direction role in. 

I have a set of 127 glyphs which is the "concept" tab here:

https://docs.google.com/spreadsheets/d/1M6SmumspyGywTigwjJcumAZT5bVSPEBH8iODEkdfk9Y/edit#gid=0

That doc also has a few other glyph sets marked up in various ways. 

So, I agree that there is a "kernel" glyph set that is to be always required under any circumstances and we should by policy consider the font buggy when it doesn't contain those glyphs. 

Then there's ought to be a 'foundation' set which should be supported by the majority of multi-script projects, since nations with majority languages in Latin have important minority communities using other scripts; an Arab living in Warsaw or a Tamilian living in Portugal ought to be able to have at hand support for their local majority language when using a font that was made primarily for their own script.

I see the current 'core' set as a compromise between those 2 needs, which Raph made over 10 years ago during the sprint to launch Google Fonts.

I've shared edit access to that Google Sheet with the participants in this thread so far, and I've added 2 new tabs for us to develop these 2 new sets, Kernel and Foundation.

**Comment by davelab6** [comment]
Thanks for drafting ideas :) Comments welcome!

**Comment by davelab6** [comment]
https://twitter.com/aaronbell/status/1438978867977285635 

I wonder where we should include characters needed for North American Latin

**Comment by thlinard** [comment]
> https://twitter.com/aaronbell/status/1438978867977285635
> 
> I wonder where we should include characters needed for North American Latin

I think that each character defined by Unicode has its uses, deserves to be represented and to appear by a glyph in a font.

But GF Latin Core Set must be is sort of a triage, right? It defines priorities according to available resources: like triage in the hospital, we want everyone to be able to receive acute care, which cannot be provided for lack of means, so we prioritize (the long-term solution being to allocate more resources, which for a character set means making it bigger and requiring fonts to have more language coverage).

I initially thought the goal was to recreate Adobe Latin 2 with comb diatrictics, which would be a pretty conservative choice. But if the idea is to start afresh on new and rational bases, depending on the number of speakers as Adam suggested, so much the better. And if you also want to raise the bar by creating a large Core Set, I applaud!
